### PR TITLE
chore: update to josdk v5.0.0-beta1 and fabric8-client v7.0.0 and related fixes

### PR DESCRIPTION
**Pull request by ebma16**
Update josdk-spring-boot-starter for compatibility with josdk v5.0.0-beta1 and fabric8 kubernetes-client v7.0.0

### CODE REVIEW

File: starter/pom.xml [hunk 1]
```diff
@@ -115,5 +115,11 @@
       <artifactId>jenvtest-fabric8-client-support</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
```
Comment (csviri) [inline starter/pom.xml:118]: Why this added?

Comment (ebma16) [reply starter/pom.xml:118]: I had some errors when building the project locally, as the import of [jetbrains annotations](https://github.com/operator-framework/josdk-spring-boot-starter/blob/9d69dd0ce12e60fe8efea87d72b56e8db02a34a6/starter/src/main/java/io/javaoperatorsdk/operator/springboot/starter/CRDApplier.java#L10) was not working without this dependency.

Comment (csviri) [reply starter/pom.xml:118]: could you try to remove it and see if it fails on CI here?

Comment (csviri) [reply starter/pom.xml]: https://github.com/ebma16/josdk-spring-boot-starter/blob/cbbf449022cdeb2fba8e39da45af2a478e77bf88/starter/src/main/java/io/javaoperatorsdk/operator/springboot/starter/CRDApplier.java#L30

Can you remove the annotation pls from here?

Comment (ebma16) [reply starter/pom.xml]: annotation removed -> CI is successful again

**Review by csviri — APPROVED** [review]
LGTM thank you!

**Pull request merged**